static-check/src/objects/symbol_table.rs

```diff
@@ -0,0 +1,492 @@
+
+            // Create new circuit type for symbol.
+            let circuit_type = CircuitType::new(&self, circuit.to_owned())?;
+
+            // Insert circuit type of symbol.
+            self.insert_circuit_type(circuit_type.identifier.clone(), circuit_type);
+        }
+
+        Ok(())
+    }
+
+    ///
+    /// Inserts one or more imported symbols for a given imported package.
+    ///
+    /// Checks that the package and all circuit and function names exist. Checks that imported circuit
+    /// and function types only contain known types.
+    ///
+    pub fn check_package(
+        &mut self,
+        import: &ImportStatement,
+        import_parser: &ImportParser,
+    ) -> Result<(), SymbolTableError> {
+        // Get imported symbols from statement.
+        let imported_symbols = ImportedSymbols::new(import);
+
+        // Import all symbols from an imported file for now.
+        // Keep track of which import files have already been checked.
+        let mut checked = Vec::new();
+
+        // Iterate over each imported symbol.
+        for (name, symbol) in imported_symbols.symbols {
+            // Skip the imported symbol if we have already checked the file.
+            if checked.contains(&name) {
+                continue;
+            };
+
+            // Find the imported program.
+            let program = import_parser
+                .get_import(&name)
+                .ok_or_else(|| SymbolTableError::unknown_package(&name, &symbol.span))?;
+
+            // Check the imported program for duplicate types.
+            self.check_program_names(program, import_parser)?;
+
+            // Check the imported program for undefined types.
+            self.check_types_program(program)?;
+
+            // Push the imported file's name to checked import files.
+            checked.push(name);
+
+            // Store the imported symbol.
+            // self.insert_import_symbol(symbol, program)?; // TODO (collinc97) uncomment this line when public/private import scopes are implemented.
+        }
+
+        Ok(())
+    }
+
+    ///
+    /// Inserts the imported symbol into the symbol table if it is present in the given program.
+    ///
+    pub fn insert_import_symbol(&mut self, symbol: ImportSymbol, program: &Program) -> Result<(), SymbolTableError> {
+        // Check for import *.
+        if symbol.is_star() {
+            // Insert all program circuits.
+            self.check_circuit_names(&program.circuits)?;
+
+            // Insert all program functions.
+            self.check_function_names(&program.functions)
+        } else {
+            // Check for a symbol alias.
+            let identifier = symbol.alias.to_owned().unwrap_or(symbol.symbol.to_owned());
+
+            // Check if the imported symbol is a circuit
+            let matched_circuit = program
+                .circuits
+                .iter()
+                .find(|(circuit_name, _circuit_def)| symbol.symbol == **circuit_name);
```

Comment:
aren't `circuits` in a `HashMap`? unless I'm missing anything, `get` should be used here instead